core/include/sched.h

```diff
@@ -278,6 +278,69 @@ typedef void (*sched_callback_t)(kernel_pid_t active, kernel_pid_t next);
 void sched_register_cb(sched_callback_t callback);
 #endif /* MODULE_SCHED_CB */
 
+#if IS_USED(MODULE_SCHED_RUNQUEUE_API) || defined(DOXYGEN)
+/**
+ * @brief   Scheduler runqueue callback
+ *
+ * @param   prio      the priority of the runqueue that changed
+ *
+ * @warning This API is not intended for out of tree users.
+ *          Breaking API changes will be done without notice and
+ *          without deprecation. Consider yourself warned!
+ */
+typedef void (*sched_runqueue_callback_t)(uint8_t prio);
+
+/**
+ * @brief  Register a callback that will be called on every runqueue change
+ *
+ * @param[in] callback The callback functions that will be called
+ *
+ * @warning This API is not intended for out of tree users.
+ *          Breaking API changes will be done without notice and
+ *          without deprecation. Consider yourself warned!
+ */
+void sched_register_runqueue_cb(sched_runqueue_callback_t callback);
+#endif /*MODULE_SCHED_RUNQUEUE_API*/
+
+/**
+ * @brief   round robin a runqueue
+ *
+ * @param   prio      the priority of the runqueue to round robin
+ *
+ * @warning This API is not intended for out of tree users.
+ *          Breaking API changes will be done without notice and
+ *          without deprecation. Consider yourself warned!
+ */
+static inline void sched_round_robin_runqueue(uint8_t prio)
+{
+    clist_lpoprpush(&sched_runqueues[prio]);
+}
+
+/**
+ * @brief   count the number of threads in a runqueue up to 2
+ *
+ * @param[in]   prio      the priority of the runqueue to get the length from
+ * @returns     number of threads in a runqueue up to 2
+ * @retval 0    The runqueue is empty
+ * @retval 1    There is exactly one thread in the runqueue
+ * @retval 2    There are at least two threads in the runqueue
+ *
+ * @warning This API is not intended for out of tree users.
+ *          Breaking API changes will be done without notice and
+ *          without deprecation. Consider yourself warned!
+ */
+static inline uint_fast8_t sched_runqueue_len2(uint8_t prio)
+{
+    uint_fast8_t c = 0;
+    if (sched_runqueues[prio].next) {
```

Comment:
```suggestion
    uint_fast8_t c = 0;

    if (sched_runqueues[prio].next) {
```

uncrustify insists on a line break here. I have no idea why this is mandatory, but I personally think it increases readability.

Reply:
It is configured to enforce an empty line after the "variable declaration block" in a function

Reply:
i am no sure if a single declaration can be called a block, but i added that empty line